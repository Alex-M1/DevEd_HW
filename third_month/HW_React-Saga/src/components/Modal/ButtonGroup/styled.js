import styled from 'styled-components';

export const StBtnGroup = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
