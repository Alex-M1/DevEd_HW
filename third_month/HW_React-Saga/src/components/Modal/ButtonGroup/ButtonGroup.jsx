import React from 'react';
import PropTypes from 'prop-types';
import Button from '../../__common__/Button/Button';
import { StBtnGroup } from './styled';

const ButtonGroup = ({ postPerson, closeModal }) => (
  <StBtnGroup>
    <Button type="primary" onClick={postPerson}>Отправить</Button>
    <Button type="cancel" onClick={closeModal}>Отменить</Button>
  </StBtnGroup>
);

export default ButtonGroup;

ButtonGroup.propTypes = {
  closeModal: PropTypes.func.isRequired,
  postPerson: PropTypes.func.isRequired,
};
