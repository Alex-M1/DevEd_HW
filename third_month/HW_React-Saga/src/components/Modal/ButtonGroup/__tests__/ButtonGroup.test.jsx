import React from 'react';
import ButtonGroup from '../ButtonGroup';

describe('ButtonGroup', () => {
  const props = {
    postPerson: jest.fn(),
    closeModal: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = global.shallowSmart(<ButtonGroup {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StBtnGroup', () => {
    const component = global.mountSmart(<ButtonGroup {...props} />);
    expect(component.find('styled__StBtnGroup')).toHaveLength(1);
  });
  it('Should have Button', () => {
    const component = global.mountSmart(<ButtonGroup {...props} />);
    expect(component.find('Button')).toHaveLength(2);
  });
  it('Should have postTheme', () => {
    const component = global.mountSmart(<ButtonGroup {...props} />);
    const sendButton = component.find('Button').at(0);
    sendButton.simulate('click');
    expect(component.props().postPerson).toHaveBeenCalled();
  });
  it('Should have close', () => {
    const component = global.mountSmart(<ButtonGroup {...props} />);
    const closeButton = component.find('Button').at(1);
    closeButton.simulate('click');
    expect(component.props().closeModal).toHaveBeenCalled();
  });
});
