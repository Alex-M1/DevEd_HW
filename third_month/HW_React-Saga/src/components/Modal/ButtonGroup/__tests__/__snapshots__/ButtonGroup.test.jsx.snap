// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Should match snapshot 1`] = `"<div class=\\"styled__StBtnGroup-c54mt-0 RBvzV\\"><button mode=\\"primary\\" class=\\"styled__StButton-iorpwz-0 GXmVc\\">Отправить</button><button mode=\\"cancel\\" class=\\"styled__StButton-iorpwz-0 nuamV\\">Отменить</button></div>"`;
